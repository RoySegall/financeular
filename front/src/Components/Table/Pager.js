import {Next, Prev} from "../Icons/Icons";
import React from "react";

export default ({prevPageCallback, currentPage, numberOfPages, nextPageCallback}) => <span>
  <button onClick={prevPageCallback} className="pr-1 cursor-pointer underline"><Prev/></button>
  {currentPage + 1} of {numberOfPages}
  <button onClick={nextPageCallback} className="pl-1 cursor-pointer underline"><Next/></button>
</span>
