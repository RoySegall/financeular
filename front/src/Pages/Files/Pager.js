import {Next, Prev} from "../../Components/Icons/Icons";
import React from "react";

export default ({pages, borderColor, currentPage, setCurrentPage, numberOfPages}) => <ul className="flex float-right">
  <li className={`p-2 border border-${borderColor}-400 bg-${borderColor}-50 text-${borderColor}-700 border-r-0 rounded-l-lg`}>
    <button onClick={() => {currentPage > 0 && setCurrentPage(currentPage - 1)}}><Prev /></button>
  </li>

  {[...Array(pages)].map((_, i) => <li key={i} className={`p-2 border border-${borderColor}-400 bg-${borderColor}-50  border-r-0 cursor-pointer`}>
    <button onClick={() => setCurrentPage(i)} className={i === currentPage ? 'font-semibold text-black' : `text-${borderColor}-700`}>{i + 1}</button>
  </li>)}

  <li className={`p-2 border border-${borderColor}-400 bg-${borderColor}-50 text-${borderColor}-700 rounded-r-lg`}>
    <button onClick={() => {currentPage + 1 < numberOfPages && setCurrentPage(currentPage + 1)}}><Next /></button>
  </li>
</ul>
