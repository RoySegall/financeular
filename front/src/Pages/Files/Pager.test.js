import renderer from "react-test-renderer";
import React from "react";
import Pager from "./Pager";
import {clickOnElement, elementShouldContainText, mountComponent} from "../../Tests/Utils";

const mockSetCurrentPage = jest.fn();

describe('Pager component', () => {

  it('Snap shot the component', () => {
    const tree = renderer
      .create(<Pager
        borderColor={'green'}
        currentPage={0}
        setCurrentPage={() => {}}
        numberOfPages={20}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Testing the methods are being invoked by a cretin state', () => {
    const wrapper = mountComponent({
      component: <Pager
        borderColor={'green'}
        currentPage={0}
        setCurrentPage={mockSetCurrentPage}
        numberOfPages={3}/>,
      wrapWithProvider: false
    });

    elementShouldContainText(wrapper.find('.font-semibold.text-black'), 1);

    // Hit the prev page anv verify the number still is 1.
    clickOnElement(wrapper.find('.prev-button'));
    elementShouldContainText(wrapper.find('.font-semibold.text-black'), 1);
    expect(mockSetCurrentPage).toBeCalledTimes(0);

    // Click on the next button and verify the callback was triggered.
    clickOnElement(wrapper.find('.next-button'));
    expect(mockSetCurrentPage).toBeCalledTimes(1);

    // Change the prop and verify the second item is selected.
    wrapper.setProps({ currentPage: 1 });
    elementShouldContainText(wrapper.find('.font-semibold.text-black'), 2);

    // Clicking the next button, because this is still available.
    clickOnElement(wrapper.find('.next-button'));
    expect(mockSetCurrentPage).toBeCalledTimes(2);

    // Set the active page as the last item and verify the next button won't
    // work.
    wrapper.setProps({ currentPage: 2 });

    clickOnElement(wrapper.find('.next-button'));
    expect(mockSetCurrentPage).toBeCalledTimes(2);

    // Hit the prev button and verify the callback was triggered.
    clickOnElement(wrapper.find('.prev-button'));
    expect(mockSetCurrentPage).toBeCalledTimes(3);
  });

});
