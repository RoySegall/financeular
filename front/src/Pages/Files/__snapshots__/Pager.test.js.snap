// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager component Snap shot the component 1`] = `
<ul
  className="flex float-right pager"
>
  <li
    className="p-2 border border-green-400 bg-green-50 text-green-700 border-r-0 rounded-l-lg"
  >
    <button
      className="prev-button"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-double-left fa-w-14 "
        data-icon="angle-double-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="font-semibold text-black"
      onClick={[Function]}
    >
      1
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      2
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      3
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      4
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      5
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      6
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      7
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      8
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      9
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      10
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      11
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      12
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      13
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      14
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      15
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      16
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      17
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      18
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      19
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50  border-r-0 cursor-pointer"
  >
    <button
      className="text-green-700"
      onClick={[Function]}
    >
      20
    </button>
  </li>
  <li
    className="p-2 border border-green-400 bg-green-50 text-green-700 rounded-r-lg"
  >
    <button
      className="next-button"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-double-right fa-w-14 "
        data-icon="angle-double-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </li>
</ul>
`;
